provider/execution/execution.go

```diff
@@ -364,3 +366,51 @@ func (e TableExecutor) handleResolveError(meta schema.ClientMeta, err error, opt
 		WithSummary("failed to resolve table \"%s\"", e.Table.Name)}, opts...)
 	return FromError(err, opts...)
 }
+
+func optResourceId(r *schema.Resource) []Option {
+	if r == nil {
+		return nil
+	}
+
+	return []Option{
+		WithResourceID(strings.Join(r.PrimaryKeyValues(), ",")),
+	}
+}
+
+func wrapResourceId(r *schema.Resource, err error) error {
+	if r == nil {
+		return err
+	}
+
+	// don't double wrap
+	var prev *errorWithResourceId
+	if errors.As(err, &prev) {
+		return err
+	}
+
+	return &errorWithResourceId{
+		err:        err,
+		resourceId: r.PrimaryKeyValues(),
```

Comment:
note that some PK values are *string etc' so the []interface{} can return [&0x14213821, ...] which is unreadable, I solved this partially but its not 100%.